Actions/ReadSecrets/ReadSecrets.ps1

```diff
@@ -128,20 +163,24 @@ try {
     }
     if ($secretsCollection) {
         Write-Host "The following secrets was not found: $(($secretsCollection | ForEach-Object { 
-            $secretSplit = @($_.Split('='))
-            if ($secretSplit.Count -eq 1) {
-                $secretSplit[0]
-            }
-            else {
-                "$($secretSplit[0]) (Secret $($secretSplit[1]))"
-            }
-            $outSecrets += @{ ""$($secretSplit[0])"" = """" }
-        }) -join ', ')"
+        $secretSplit = @($_.Split('='))
+        if ($secretSplit.Count -eq 1) {
+            $secretSplit[0]
+        }
+        else {
+            "$($secretSplit[0]) (Secret $($secretSplit[1]))"
+        }
+        $outSecrets += @{ ""$($secretSplit[0])"" = """" }
+    }) -join ', ')"
     }
 
     $outSecretsJson = $outSecrets | ConvertTo-Json -Compress
-    Add-Content -Path $env:GITHUB_ENV -Value "RepoSecrets=$OutSecretsJson"
+    Add-Content -Path $env:GITHUB_ENV -Value "RepoSecrets=$outSecretsJson"
+
+    $outSettingsJson = $outSettings | ConvertTo-Json -Compress
+    Add-Content -Path $env:GITHUB_ENV -Value "Settings=$OutSettingsJson"

```

Comment:
The code here adds the settings to ENV and overrides the settings set by readsettings. I think that is fine.
You don't however set the output value, by writing this to host ::set-output -name=settings::settingsinjson
This means that the output defined in yaml is never set and you cannot reference the settings through the outputs.
ENV variables only works inside the same job, referencing the outputs through outputs is useful when using settings and secrets across jobs, one job can define the output of an action as the output from the job.
Right now, you don't use that as you use env:settings in runpipeline with.
I think you should set the output value here as well and use that and I actually think that both readsettings and readsecrets should stop emitting the full combined settings as ENV - and readsecrets should emit reposecrets as outputs as well instead of ENV. (sorry for being the one pulling you down the ENV path:-))